From: Vikas Chaudhary <vikas.chaudhary@qlogic.com>
Subject: Update qla4xxx to v5.01.00.00.11.01-k12
References: bnc#556572
Patch-Mainline: Submitted to scsi-misc

- Remove double unlock in qla4xxx_eh_abort()
  In qla4xxx_eh_abort hardware lock was unlocked twice, should be done once.

- Move debug message to print correct status of sp reference count.

Signed-off-by: Ravi Anand <ravi.anand@qlogic.com>
Acked-by: Hannes Reinecke <hare@suse.de>

diff --git a/drivers/scsi/qla4xxx/ql4_os.c b/drivers/scsi/qla4xxx/ql4_os.c
index 429d068..4d72a88 100644
--- a/drivers/scsi/qla4xxx/ql4_os.c
+++ b/drivers/scsi/qla4xxx/ql4_os.c
@@ -1925,7 +1925,6 @@ static int qla4xxx_eh_abort(struct scsi_cmnd *cmd)
 	spin_unlock_irqrestore(&ha->hardware_lock, flags);
 
 	/* Wait for command to complete */
-	spin_unlock_irqrestore(&ha->hardware_lock, flags);
 	if (qla4xxx_eh_wait_on_command(ha, cmd, got_ref)) {
 		dev_info(&ha->pdev->dev,
 			"scsi%ld:%d:%d:%d: ABORT SUCCEEDED - "
@@ -1934,13 +1933,13 @@ static int qla4xxx_eh_abort(struct scsi_cmnd *cmd)
 		ret = SUCCESS;
 	}
 
-	if (got_ref)
-		sp_put(ha, srb);
-
 	DEBUG2(printk("scsi%ld:%d:%d:%d: ABORT cmd=%p, pid=%ld, ref=%d, "
 		      "ret=%x\n", ha->host_no, channel, id, lun, cmd,
 		      serial, atomic_read(&srb->ref_count), ret));
 
+	if (got_ref)
+		sp_put(ha, srb);
+
 	return ret;
 }
 
diff --git a/drivers/scsi/qla4xxx/ql4_version.h b/drivers/scsi/qla4xxx/ql4_version.h
index b2eb46e..cdef579 100644
--- a/drivers/scsi/qla4xxx/ql4_version.h
+++ b/drivers/scsi/qla4xxx/ql4_version.h
@@ -5,5 +5,5 @@
  * See LICENSE.qla4xxx for copyright and licensing details.
  */
 
-#define QLA4XXX_DRIVER_VERSION	"5.01.00.00.11.01-k11"
+#define QLA4XXX_DRIVER_VERSION	"5.01.00.00.11.01-k12"
 
